### PR TITLE
ARROW-3196: Add support for merging both ARROW and PARQUET patches

### PR DESCRIPTION
**Pull request by wesm**
I will test this out on the PARQUET-1398 patch. We should make sure we can merge both ARROW and PARQUET patches before merging this

### CODE REVIEW

**Comment by wesm** [comment]
I merged PARQUET-1398. There's still a little work to do on specifying different fix versions, so I'll keep working at this. I will probably write some unit tests for the merge script

**Comment by guoyuhong** [comment]
[`arrow::TypePtr`](https://github.com/apache/parquet-cpp/blob/master/src/parquet/arrow/schema.cc#L32) is removed from Arrow but it is still referred from Parquet master. There will be building failures if we use recent Arrow as will as Parquet. Will this also be fixed in this PR? ;-)

**Comment by wesm** [comment]
@guoyuhong https://github.com/apache/parquet-cpp is now defunct, please remove this from your toolchain and add `-DARROW_PARQUET=ON` when building Arrow

**Comment by wesm** [comment]
See https://github.com/apache/parquet-cpp/commit/93683ce0b6681dcab1c023e665c1999ef0715d50

**Comment by wesm** [comment]
@pcmoritz @robertnishihara could you let me know what is the status of moving Ray to use the PyPI packages?

**Comment by guoyuhong** [comment]
@wesm Thanks for the information. I have submitted a PR to change Ray's building process.

**Comment by robertnishihara** [comment]
@wesm, there's no update about this, though we could try to make the next Ray release (0.6) use the next Arrow release.

**Comment by wesm** [comment]
OK, we should be releasing pretty soon so that would be great!

**Comment by wesm** [comment]
This might be a bit elaborate but I was getting tired of the brittleness of this script. It'd be really difficult to test the stuff where git is involved so I skipped that. I'm going to use this to merge some more PRs before merging this back to the repo but this moves things in the right direction, I think

This also includes the fix for ARROW-3116

File: dev/merge_arrow_pr.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 # Utility for creating well-formed pull request merges and pushing them to
 # Apache.
-#   usage: ./apache-pr-merge.py    (see config env vars below)
+#   usage: ./merge_arrow_py.py    (see config env vars below)
```
Comment (robertnishihara) [inline dev/merge_arrow_pr.py]: spelling I think

**Comment by wesm** [comment]
OK, this is working fine now. I'm merging and will fix any more issues that come up

**Comment by wesm** [comment]
+1

**Pull request closed without merging**